monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx

```diff
@@ -0,0 +1,186 @@
+      setSuccessfullyUninstalledPluginsIds((prevState) => {
+        return shallowAdditionOfUniqueValueToArray(prevState, pluginId);
+      });
+      refreshInstalledPlugins();
+    }).catch(() => {
+      console.log('error uninstalling plugin');
+    }).finally(() => {
+      setPluginsInUninstallProcess((prevState => {
+        return shallowRemovalOfUniqueValueFromArray(prevState, pluginId);
+      }));
+    });
+  }
+
+  const upgradePlugin = (pluginType, name, version) => {
+    const options = {
+      method: 'PUT',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({plugin_type: pluginType, name: name, version: version})
+    }
+    return authComponent.authFetch('/api/install-agent-plugin', options, true);
+  }
+
+  const onUpgradeClick = (pluginId, pluginType, name, version) => {
+    setPluginsInInstallProcess((prevState) => {
+      return shallowAdditionOfUniqueValueToArray(prevState, pluginId);
+    });
+
+    upgradePlugin(pluginType, name, version).then(() => {
+      setSuccessfullyInstalledPluginsIds((prevState) => {
+        return shallowAdditionOfUniqueValueToArray(prevState, pluginId);
+      });
+      refreshInstalledPlugins();
+    }).catch(() => {
+      console.log('error upgrading plugin');
+    }).finally(() => {
+      setPluginsInInstallProcess((prevState => {
+        return shallowRemovalOfUniqueValueFromArray(prevState, pluginId);
+      }));
+    });
+  }
+
+  const getUpgradeAction = (row) => {
+    const pluginId = row.id;
+    if (pluginsInInstallProcess.includes(pluginId)) {
+      return [
+        <GridActionsCellItem
+          key={pluginId + 'upgrade'}
+          icon={<DownloadingIcon/>}
+          label="Upgrading"
+          className="textPrimary"
+          color="inherit"
+        />
+      ]
+    }
+
+    if (successfullyInstalledPluginsIds.includes(pluginId)) {
+      return [
+        <GridActionsCellItem
+          key={pluginId + 'upgrade'}
+          icon={<DownloadDoneIcon/>}
+          label="Upgrade Complete"
+          className="textPrimary"
+          color="inherit"
+        />
+      ]
+    }
+
+    if (row.update_version) {
+      return [
+        <GridActionsCellItem
+        key={pluginId + 'upgrade'}
+        icon={<UpgradeIcon/>}
+        label="Upgrade"
+        className="textPrimary"
+        onClick={() => onUpgradeClick(pluginId, row.type, row.name, row.update_version)}
+        color="inherit"
+      />
+      ]
+    }
+
+    return [
+      <GridActionsCellItem
+      key={pluginId + 'upgrade'}
+      icon={<UpgradeIcon/>}
+      label="Upgrade"
+      className="textPrimary"
+      disabled={true}
+      color="inherit"
+    />
+    ];
+  }
+
+  const getUninstallAction = (row) => {
+    const pluginId = row.id;
+    if (pluginsInUninstallProcess.includes(pluginId)) {
+      return [
+        <GridActionsCellItem
+          key={pluginId + 'uninstall'}
+          icon={<DownloadingIcon/>}
```

Comment:
maybe this icon is better?
import AutoDeleteIcon from '@mui/icons-material/AutoDelete';

your decision

Reply:
Looks good to me